### PR TITLE
Use model factory in gui

### LINKED ISSUE #549: Refactor `SimulationPanel` to use shared default settings
Analysis modules are getting default values differently through the `CLI` and the `GUI`, the `GUI`should be refactored to use defaults from the `model_factory`.

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #549 

**Approach**
_Short description of the approach_

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by ManInFez** [comment]
Test ert please!

**Comment by ManInFez** [comment]
test ert please!

**Comment by ManInFez** [comment]
test ert please!

File: ert_gui/simulation/ensemble_experiment_panel.py [hunk 1]
```diff
@@ -15,6 +15,10 @@
 from ert_gui.simulation.simulation_config_panel import SimulationConfigPanel
 from ert_shared.models import EnsembleExperiment
 
+from collections import namedtuple
+
+Arguments = namedtuple("Arguments", ["mode", "iter_num", "realizations"])
```
Comment (oyvindeide) [inline ert_gui/simulation/ensemble_experiment_panel.py:20]: How about using `@dataclass` instead? Personally I find it easier to read:
```suggestion
@dataclass
class Arguments:
    mode: str
    iter_num: int
    realisations: str
```
?

Comment (ManInFez) [reply ert_gui/simulation/ensemble_experiment_panel.py:20]: I will give it a go

Comment (ManInFez) [reply ert_gui/simulation/ensemble_experiment_panel.py]: Used @dataclass, I agree it looks more readable

File: ert_gui/simulation/multiple_data_assimilation_panel.py [hunk 2]
```diff
@@ -37,6 +37,17 @@
...
 
+from dataclasses import dataclass
+
+
+@dataclass
+class Arguments:
+    mode: str
+    target_case: str
+    realizations: str
+    weights: list
```
Comment (oyvindeide) [inline ert_gui/simulation/multiple_data_assimilation_panel.py:48]: Could do `List[float]` (or is it `List[int]`) here

**Review by oyvindeide — APPROVED** [review]
LGTM!

**Pull request merged**